lib/deref/index.js

```diff
@@ -1,61 +1,57 @@
-var Rx = require("rx/dist/rx");
-var pathSyntax = require("falcor-path-syntax");
+var __key = require("./../internal/key");
+var __path = require("./../internal/path");
+var __parent = require("./../internal/parent");
+var InvalidDerefInputError = require("./../errors/InvalidDerefInputError");
 
-module.exports = function deref(boundPathArg) {
+module.exports = function deref(boundJSONArg) {
+    var path = boundJSONArg && boundJSONArg[__path];
+    var key = boundJSONArg && boundJSONArg[__key];
```

Comment:
We could move the cost of this lookup to after the early path return (the invalid input error would need to be moved down also), but gives the fastest path out of deref.